src/xdist/remote.py

```diff
@@ -116,6 +116,20 @@ def run_one_test(self, torun):
             "runtest_protocol_complete", item_index=self.item_index, duration=duration
         )
 
+    def pytest_collection_modifyitems(self, session, config, items):
+        # add the group name to nodeid as suffix if --dist=loadgroup
+        if config.getvalue("loadgroup"):
+            for item in items:
+                try:
+                    mark = item.get_closest_marker("xgroup")
+                except AttributeError:
+                    mark = item.get_marker("xgroup")
```

Comment:
We only support pytest>=6.2 now, so this can be changed to:

```suggestion
                mark = item.get_closest_marker("xgroup")
```

Reply:
Thank you. 62e50d0